pkg/parse/parse.go

```diff
@@ -649,23 +649,27 @@ func getAuthFile(authfile string) string {
 // PlatformFromOptions parses the operating system (os) and architecture (arch)
 // from the provided command line options.
 func PlatformFromOptions(c *cobra.Command) (os, arch string, err error) {
-	os = runtime.GOOS
-	arch = runtime.GOARCH
 
-	if selectedOS, err := c.Flags().GetString("os"); err == nil && selectedOS != runtime.GOOS {
-		os = selectedOS
+	if c.Flag("os").Changed {
```

Comment:
- This does not change `SystemContextFromOptions`
- From a very quick skim, the results of _this_ function are used to set the OS/arch inside the built image, and not passed to c/image at all. That probably should have _some_ default.
- It seems easier to follow to me to not have the `runtime.GO*` defaults for flag values (in `cmd/buildah/`) specified, if they are going to be ignored here, and if this function hard-codes a default anyway.

Disclaimer: This is based on a very cursory skim, I don’t understand the code base and data flow at all well.

Reply:
The advantage of the defaults is the buildah --help information shows them

$ ./bin/buildah from --help | grep -- --os
      --os OS                         prefer OS instead of the running OS when pulling images (default "linux")
$ ./bin/buildah from --help | grep -- --arch
      --arch string                   set the ARCH of the image to the provided value instead of the architecture of the host (default "amd64")

The SystemContextFromOptions is a good point, and I added the .Changed check there as well.

Reply:
Well, but it’s _not_ the default if no `--os` and `--os=$default` behave differently.

It is also possible to embed that value into the flag description instead.

---

The change of the default return value of `PlatformFromOptions` still looks wrong to me, but I’ll leave that to Buildah experts.